src/clients/PuchasesClient.ts

```diff
@@ -0,0 +1,86 @@
+import { IPurchasesClient } from "../interfaces/purchases.interfaces";
+import { ApiCustomerPurchases, ApiCustomerSummary, ClientOptions, GetCustomerPurchasesCallback, GetCustomersBulkInfoCallback, GetCustomersBulkInfoParams, Paginated } from "../types";
+import { BaseClient } from "./BaseClient.abstract";
+
+/**
+ * class PurchasesClient
+ * constructor requires options as
+ * {
+ *  protocol: 'http',
+ *  hostname: 'domain.com',
+ *  port: 80,
+ *  pathnamePrefix?: '/v2/customers',
+ *  clientId?: 'my-client-id',
+ *  secret?: 'secret key if exists',
+ *  agent?: string
+ * }
+ */
+export class PurchasesClient extends BaseClient implements IPurchasesClient {
+
+  /**
+   * constructor
+   * @param options ClientOptions
+   */
+  constructor(options: ClientOptions) {
+    super(`${options.protocol}://${options.hostname}:${options.port}${options.pathnamePrefix}`, options);
+
+  }
+
+  /**
+   * Get customer purchases
+   * @param applicationUsername : string application-username
+   * @param cb : callback method for getting the data.
+   * @returns : void
+   */
+  getCustomerPurchases(applicationUsername: string, cb: GetCustomerPurchasesCallback): void {
+
+    if (applicationUsername === null || applicationUsername === '' || applicationUsername === undefined) {
+      return cb(new Error("UserId parameter is required to be non-null and non-empty"));
+    }
+
```

Comment:
I know this is redundant in typescript, but checking the type could be good too (because someone might be using this lib from javascript).

```ts
if (typeof applicationUsername !== 'string') return cb(new Error("..."));
```

*Note: since `typeof null` and `typeof undefined` are not equal to `string`, the related tests can be removed from above too*